### ISSUE TITLE
When cancelling campaign creation a 500 error is thrown

### ISSUE THREAD

**Issue by RCheesley** [mautic/mautic#11181]
### Mautic Version

4.3.x series (current 4.x branch)

### PHP version

7.4.29

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When creating a campaign and then subsequently cancelling, a 500 error is being thrown:

`500 Internal Server Error - Unable to generate a URL for the named route "c" as such route does not exist.`

Stack trace:

```
Symfony\Component\Routing\Exception\RouteNotFoundException
/vendor/symfony/routing/Generator/CompiledUrlGenerator.php:50 at
/vendor/symfony/routing/Router.php:252 at Symfony\Component\Routing\Generator\CompiledUrlGenerator -> generate ( 'c', array('page' => '1'), '1' )
/vendor/symfony/framework-bundle/Controller/ControllerTrait.php:80 at Symfony\Component\Routing\Router -> generate ( 'c', array('page' => '1'), '1' )
/app/bundles/CoreBundle/Controller/AbstractStandardFormController.php:32 at Symfony\Bundle\FrameworkBundle\Controller\Controller -> generateUrl ( 'c', array('page' => '1'), '1' )
/app/bundles/CampaignBundle/Controller/CampaignController.php:345 at Mautic\CoreBundle\Controller\AbstractStandardFormController -> generateUrl ( 'c', array('page' => '1') )
/app/bundles/CoreBundle/Controller/CommonController.php:447 at Mautic\CampaignBundle\Controller\CampaignController -> newAction ( '0', '' )
/vendor/symfony/http-kernel/HttpKernel.php:158 at Mautic\CoreBundle\Controller\CommonController -> executeAction ( 'new', '0', '0', '' )
/vendor/symfony/http-kernel/HttpKernel.php:80 at Symfony\Component\HttpKernel\HttpKernel -> handleRaw ( object(Request), '1' )
/vendor/symfony/http-kernel/Kernel.php:201 at Symfony\Component\HttpKernel\HttpKernel -> handle ( object(Request), '1', true )
/app/AppKernel.php:115 at Symfony\Component\HttpKernel\Kernel -> handle ( object(Request), '1', true )
/app/middlewares/CORSMiddleware.php:82 at AppKernel -> handle ( object(Request), '1', true )
/app/middlewares/CatchExceptionMiddleware.php:34 at Mautic\Middleware\CORSMiddleware -> handle ( object(Request), '1', true )
/app/middlewares/Dev/IpRestrictMiddleware.php:52 at Mautic\Middleware\CatchExceptionMiddleware -> handle ( object(Request), '1', true )
/app/middlewares/VersionCheckMiddleware.php:58 at Mautic\Middleware\Dev\IpRestrictMiddleware -> handle ( object(Request), '1', true )
/app/middlewares/TrustMiddleware.php:42 at Mautic\Middleware\VersionCheckMiddleware -> handle ( object(Request), '1', true )
/vendor/stack/builder/src/Stack/StackedHttpKernel.php:23 at Mautic\Middleware\TrustMiddleware -> handle ( object(Request), '1', true )
/vendor/stack/run/src/Stack/run.php:13 at Stack\StackedHttpKernel -> handle ( object(Request) )
/index_dev.php:25 at Stack\run ( object(StackedHttpKernel) )
```

### How can we reproduce this issue?

Step 1: Create a new campaign
Step 2: Cancel creating the campaign

### Relevant log output

```shell
[2022-05-20 13:41:05] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Routing\Exception\RouteNotFoundException: "Unable to generate a URL for the named route "c" as such route does not exist." at /var/www/html/vendor/symfony/routing/Generator/CompiledUrlGenerator.php line 50 {"exception":"[object] (Symfony\\Component\\Routing\\Exception\\RouteNotFoundException(code: 0): Unable to generate a URL for the named route \"c\" as such route does not exist. at /var/www/html/vendor/symfony/routing/Generator/CompiledUrlGenerator.php:50)
[stacktrace]
#0 /var/www/html/vendor/symfony/routing/Router.php(252): Symfony\\Component\\Routing\\Generator\\CompiledUrlGenerator->generate('c', Array, 1)
#1 /var/www/html/vendor/symfony/framework-bundle/Controller/ControllerTrait.php(80): Symfony\\Component\\Routing\\Router->generate('c', Array, 1)
#2 /var/www/html/app/bundles/CoreBundle/Controller/AbstractStandardFormController.php(32): Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller->generateUrl('c', Array, 1)
#3 /var/www/html/app/bundles/CampaignBundle/Controller/CampaignController.php(345): Mautic\\CoreBundle\\Controller\\AbstractStandardFormController->generateUrl('c', Array)
#4 /var/www/html/app/bundles/CoreBundle/Controller/CommonController.php(447): Mautic\\CampaignBundle\\Controller\\CampaignController->newAction(0, '')
#5 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(158): Mautic\\CoreBundle\\Controller\\CommonController->executeAction('new', 0, 0, '')
#6 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(80): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)
#7 /var/www/html/vendor/symfony/http-kernel/Kernel.php(201): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#8 /var/www/html/app/AppKernel.php(115): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#9 /var/www/html/app/middlewares/CORSMiddleware.php(82): AppKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#10 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(34): Mautic\\Middleware\\CORSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#11 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(52): Mautic\\Middleware\\CatchExceptionMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#12 /var/www/html/app/middlewares/VersionCheckMiddleware.php(58): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#13 /var/www/html/app/middlewares/TrustMiddleware.php(42): Mautic\\Middleware\\VersionCheckMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#14 /var/www/html/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Mautic\\Middleware\\TrustMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#15 /var/www/html/vendor/stack/run/src/Stack/run.php(13): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))
#16 /var/www/html/index_dev.php(25): Stack\
un(Object(Stack\\StackedHttpKernel))
#17 {main}
"} {"hostname":"mautic-web","pid":2189}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

**Comment by TechBill** [comment]
I am experiencing the same bug on our Mautic but this bug only occur to us on new campaign (new campaign -> cancel). 

It does not give a 500 error when cancelling from existing campaign (select campaign -> edit -> cancel) which return back to campaign page.

**Comment by RCheesley** [comment]
@TechBill yes this was my situation as well.

**Comment by RCheesley** [comment]
Closing issue as it was fixed with #11348
